src/Microsoft.ML.Data/Transforms/ColumnBindingsBase.cs

```diff
@@ -14,16 +14,44 @@
 
 namespace Microsoft.ML.Data
 {
-    internal abstract class SourceNameColumnBase
+    /// <summary>
+    /// Specifies input and output column names for a transformation
+    /// </summary>
+    public class OneToOneColumn
```

Comment:
>OneToOneColumn [](start = 17, length = 14)

Is OneToOneColumn a good name? Usually `one-to-one` describes a mapping or a function, which doesn't sound a `Column`. Maybe we could try `OneToOneBind`?